src/org/sosy_lab/java_smt/solvers/bitwuzla/BitwuzlaTheoremProver.java

```diff
@@ -119,7 +123,8 @@ private boolean readSATResult(Result resultValue) throws SolverException, Interr
       return false;
     } else if (resultValue == Result.UNSAT) {
       return true;
-    } else if (resultValue == Result.UNKNOWN && shutdownNotifier.shouldShutdown()) {
+    } else if (resultValue == Result.UNKNOWN
+        && proverShutdownManager.getNotifier().shouldShutdown()) {
```

Comment:
Is it guaranteed that the manager always returns the same notifier-instance and the notifier always returns the same flag?

Reply:
Yes! The notifier is created final and only once, see here: `private final ShutdownNotifier notifier = new ShutdownNotifier(this);` The flag can not be changed once a shutdown is triggered.

Reply:
Referencing some existing source code is no "guarantee".
The guarantee would be a JavaDoc specification for the ShutdownManager-API, which is missing:
https://sosy-lab.github.io/java-common-lib/api/org/sosy_lab/common/ShutdownManager.html#getNotifier()

A user can provide his own implementation of the shutdown-manager and change its behaviour.
I really would prefer to keep a local reference to this notifier.

Reply:
There is no intention of a guarantee that `getNotifier()` always returns the same instance, but there is definitively the intention that this does not matter and that all notifiers handed out by a manager always return the shutdown state of the manager, and that `shouldShutdown()` will keep returning `true` forever after it has returned `true` once.

Since the split in separate manager and notifier classes this was not fully written down in the JavaDoc, but I improved this now: https://github.com/sosy-lab/java-common-lib/commit/108faa65ec0ec60a085232de43831bac1d453db3

Reply:
I am not sure what exactly you are referring to. But if you have multi-threaded code, then you can of course never be sure when each of the threads is scheduled in relation to the others.

Reply:
What do you mean with "synchronized way"? "synchronized" is not some magic global property, one can only synchronize against specific things.

If you want help, you need to be more specific. What is the sequence of operations that you are observing and that is unexpected/unwanted?